test/CancellationHandleTest.cpp

```diff
@@ -153,7 +153,7 @@ TEST(CancellationHandle, ensureObjectLifetimeIsValidWithoutWatchDogStarted) {
 namespace ad_utility {
 
 TEST(CancellationHandle, verifyWatchDogDoesChangeState) {
-#ifdef __APPLE__
+#if (defined(__APPLE__)) || defined(_QLEVER_NO_TIMING_TESTS)
```

Comment:
To be honest, I would simply rely on the custom macro and just set the flag for macos builds. In my opinion this makes more sense as it allows people on actual macs (outside of GH action runners) to run the tests normally

Reply:
Thanks for the feedback.
I have incorporated it, let's see if this works.